apps/codeimage/src/components/Presets/PresetUpdateDialog.tsx

```diff
@@ -0,0 +1,152 @@
+import {useI18n} from '@codeimage/locale';
+import {getRootEditorStore} from '@codeimage/store/editor';
+import {getFrameState} from '@codeimage/store/editor/frame';
+import {getTerminalState} from '@codeimage/store/editor/terminal';
+import {Preset} from '@codeimage/store/presets/types';
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogPanelContent,
+  DialogPanelFooter,
+  HStack,
+  SvgIcon,
+  Text,
+} from '@codeimage/ui';
+import {JSXElement, mergeProps, VoidProps} from 'solid-js';
+import {AppLocaleEntries} from '../../i18n';
+import CustomEditorPreview from '../CustomEditor/CustomEditorPreview';
+import {TerminalHost} from '../Terminal/TerminalHost';
+import * as styles from '../ThemeSwitcher/PresetSwitcher.css';
+import {ThemeBox} from '../ThemeSwitcher/ThemeBox';
+
+interface PresetUpdateDialogProps {
+  currentPreset: Preset['data'];
+  onConfirm: () => void;
+  onClose?: () => void;
+}
+
+export function PresetUpdateDialog(
+  props: VoidProps<PresetUpdateDialogProps>,
+): JSXElement {
+  const [t] = useI18n<AppLocaleEntries>();
+  const propsWithDefault = mergeProps({actionType: 'primary'} as const, props);
+
+  const exampleCode = 'function Code() {\n' + ' console.log()\n' + '}';
+
+  const currentEditorData = () => {
+    return {
+      frame: getFrameState().stateToPersist(),
+      terminal: getTerminalState().stateToPersist(),
+      editor: getRootEditorStore().stateToPersist(),
+    };
+  };
+
+  function PresetPreview(props: {data: Preset['data']}) {
```

Comment:
@riccardoperra I've already done a similar component with the same name 😅...

I think we can merge this MR then I will merge solving conflicts and continuing from here, wytd?